File: Branching Recursion/Python code/Level2.py
def generate_binary(n, current):
    if n == 0:
        print(current)
        return
    generate_binary(n - 1, current + "0")
    generate_binary(n - 1, current + "1")

def generate_paths(n, path):
    if n == 0:
        print(path)
        return
    generate_paths(n - 1, path + "L")
    generate_paths(n - 1, path + "R")

def generate_combinations(n, combo):
    if n == 0:
        print(combo)
        return
    generate_combinations(n - 1, combo + "Left ")
    generate_combinations(n - 1, combo + "Right ")

if __name__ == "__main__":
    n = 3
    
    print(f"1. Binary strings of length {n}:")
    generate_binary(n, "")
    
    print(f"\n2. Paths (L/R) for {n} steps:")
    generate_paths(n, "")
    
    print(f"\n3. Combinations (Left/Right) for {n} choices:")
    generate_combinations(n, "")

File: Branching Recursion/JS code/Level2.js
// 1. Print all possible binary strings of length n
function generateBinary(n, current) {
    if (n === 0) {
        console.log(current);
        return;
    }
    generateBinary(n - 1, current + "0");
    generateBinary(n - 1, current + "1");
}

// 2. Print all paths using L and R for n steps
function generatePaths(n, path) {
    if (n === 0) {
        console.log(path);
        return;
    }
    generatePaths(n - 1, path + "L");
    generatePaths(n - 1, path + "R");
}

// 3. Print all combinations of choosing left or right n times
function generateCombinations(n, combo) {
    if (n === 0) {
        console.log(combo);
        return;
    }
    generateCombinations(n - 1, combo + "Left ");
    generateCombinations(n - 1, combo + "Right ");
}

const n = 3;

console.log(`1. Binary strings of length ${n}:`);
generateBinary(n, "");

console.log(`\n2. Paths (L/R) for ${n} steps:`);
generatePaths(n, "");

console.log(`\n3. Combinations (Left/Right) for ${n} choices:`);
generateCombinations(n, "");
